src/map/script.c

```diff
@@ -8623,6 +8622,34 @@ BUILDIN(readparam) {
 	return true;
 }
 
+/*==============================================*
+ * setparam(<parameter number>, <value>{, "<Player Name>" | <account id> });
+ * Uses pc->setparam() on a specified Account ID. [Wolfie]
+ *==============================================*/
+BUILDIN(setparam)
+{
+	struct script_data *data = script_getdata(st, 2);
+	int type, val = script_getnum(st, 3);
+	struct map_session_data *sd;
+
+	if (reference_toparam(data))
+		type = reference_getparamtype(data);
+	else
+		type = script->conv_num(st, data);
+
+	if (script_hasdata(st, 4))
+		sd = (script_isstringtype(st, 4)) ? script->nick2sd(st, script_getstr(st, 4)) : script->id2sd(st, script_getnum(st, 4));
+	else
+		sd = script->rid2sd(st);
+
+	if (sd != NULL) {
+		if (pc->setparam(sd, type, val) == 0)
+			ShowWarning("buildin_setparam: Invalid parameter provided.\n");
```

Comment:
tested in-game
```
prontera,155,187,5	script	kjsdhfksjdhf	1_F_MARIA,{
	setparam asd, 100;
	end;
}
```
print error as
```
[Error]: pc_setparam: Attempted to set unknown parameter '0'.
[Warning]: buildin_setparam: Invalid parameter provided.
```
1. remove this ShowWarning, the `pc_setparam` function already has its own warning message
2. `return false;` to show which npc it came from